lib/util/package-rules/merge-confidence.ts

```diff
@@ -0,0 +1,19 @@
+import is from '@sindresorhus/is';
+import type { PackageRule, PackageRuleInputConfig } from '../../config/types';
+import { Matcher } from './base';
+
+export class MergeConfidenceMatcher extends Matcher {
+  override matches(
+    { mergeConfidenceLevel }: PackageRuleInputConfig,
+    { matchConfidence }: PackageRule
+  ): boolean | null {
+    if (is.nullOrUndefined(matchConfidence)) {
+      return null;
+    }
+    return (
+      is.array(matchConfidence) &&
+      is.nonEmptyString(mergeConfidenceLevel) &&
+      matchConfidence.includes(mergeConfidenceLevel)
```

Comment:
The mc level is fetched at the lookup phase.
fetching inside the matcher might be a better approach as this way we only fetch if the matcher is being used.
problem is that the `MatcherApi` is synchronous.
converting it to support `async` might turn into a big refactor (?).

wdyt?

Reply:
querying is done as part of `generateUpdate` function, and we will only query the api if the `mergeConfidence` matcher is used, otherwise we skip querying it.